perfkitbenchmarker/publisher.py

```diff
@@ -132,7 +132,7 @@ def AddMetadata(self, metadata, benchmark_spec):
     metadata = metadata.copy()
     metadata['perfkitbenchmarker_version'] = version.VERSION
     metadata['cloud'] = benchmark_spec.cloud
-    metadata['zones'] = ','.join(benchmark_spec.zones)
+    metadata['zones'] = ','.join(set([vm.zone for vm in benchmark_spec.vms]))
```

Comment:
I was confused by the set() wrapper until I realized it's for making them unique. May deserve a comment.